src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelper.cs

```diff
@@ -205,6 +209,25 @@ public virtual string Link(string routeName, object values)
             });
         }
 
+        private RouteValueDictionary GetValuesDictionary(UrlActionContext actionContext)
+        {
+            var dictionaryValues = actionContext.Values as IDictionary<string, object>;
+            if (dictionaryValues != null)
```

Comment:
It would be nice to also special case when `actionContext.Values` is a `RouteValueDictionary` - that can avoid the boxed enumerator.

what type is `actionContext.Values` usually?